ExtendCspHeadersWebpackPlugin.js

```diff
@@ -19,9 +19,13 @@ function ExtendCspHeadersWebpackPlugin() {
 
 function emit(compilation) {
   return new Promise(resolve => {
-    const csp = generateCsp(compilation.assets);
-    extendHeaders(compilation.assets, csp);
-    resolve();
+    const assets = compilation.assets;
+    if (!assets["_headers"] || !assets["_headersCsp.json"]) {
```

Comment:
Now that we are checking the individual assets in the caller anyways,
we should only pass the needed assets to the called functions.
This prevents duplication of the `assets[xyz]` code by making the called functions more dumb.